pkg/api/featureflags.go

```diff
@@ -22,4 +22,8 @@ const (
 	// FeatureFlagCheckAccessTestToggle is used for safely testing the new check access
 	// API in production. The toggle will be removed once the testing has been completed.
 	FeatureFlagCheckAccessTestToggle = "Microsoft.RedHatOpenShift/CheckAccessTestToggle"
+
+	// FeatureFlagBYONsg is used for indicating whether a customer subscription
+	// is registered for BYO NSG feature.
+	FeatureFlagBYONsg = "Microsoft.RedHatOpenShift/BYONsg"
```

Comment:
Since this will be an API option in future releases, we should track this in `pkg/api/openshiftcluster.go` and display it in our admin API

`pkg/api/admin/openshiftcluster.go`.  We will need conversions for both to internal and to external

Reply:
We'll need to update our powershell scripts and geneva automation as well.

Reply:
2 stories - one for powershell script updates in ARO-Scripts, the other for geneva automation updates in ARO.Pipelines (or ARO-Pipelines, if it's migrated already)

Reply:
ARO-3051 and ARO-3052 have been created for those stories.